src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index > -1) {
+      Animal.alive.splice(index, 1);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(biteObject) {
+    if (!(biteObject instanceof Carnivore) && !biteObject.hidden) {
```

Comment:
[FUNCTION NAMING] The `bite` method should check if the `biteObject` is an instance of `Herbivore` and not hidden to align with the task description. The current implementation checks if `biteObject` is not an instance of `Carnivore`, which could lead to issues if other classes besides `Herbivore` and `Carnivore` are introduced in the future.